### PR TITLE
Dublication Fixed in AnimatedValue.kt #925

### PR DESCRIPTION
**Pull request by GeorgeJoLo**
I have made the changes you suggest me and I also add a test for each class.

### CODE REVIEW

File: fxgl-animation/src/main/kotlin/com/almasb/fxgl/animation/AnimatedValue.kt [hunk 1]
```diff
@@ -102,32 +102,25 @@ class AnimatedCubicBezierPoint2D
     }
 }
 
-class AnimatedQuadBezierPoint3D
+internal class AnimatedQuadBezierPoint3D
 (val path: QuadCurve) : AnimatedValue<Point3D>(Point3D.ZERO, Point3D.ZERO) {
+	
+	private val Animated2D = AnimatedQuadBezierPoint2D(path)
```
Comment (AlmasB) [inline fxgl-animation/src/main/kotlin/com/almasb/fxgl/animation/AnimatedValue.kt:108]: Something odd with formatting, please check that spaces are set to 4 and no tabs.

Comment (AlmasB) [inline fxgl-animation/src/main/kotlin/com/almasb/fxgl/animation/AnimatedValue.kt:108]: Variable names start with lowercase, so `Animated2D -> animated2D`

File: fxgl-animation/src/main/kotlin/com/almasb/fxgl/animation/AnimatedValue.kt [hunk 2]
```diff
@@ -102,32 +102,25 @@ class AnimatedCubicBezierPoint2D
...
 
         return Point3D(p.x, p.y, 0.0)
     }
 }
 
-class AnimatedCubicBezierPoint3D
+internal class AnimatedCubicBezierPoint3D
 (val path: CubicCurve) : AnimatedValue<Point3D>(Point3D.ZERO, Point3D.ZERO) {
+	
+	private val Animated2D = AnimatedCubicBezierPoint2D(path)
```
Comment (AlmasB) [inline fxgl-animation/src/main/kotlin/com/almasb/fxgl/animation/AnimatedValue.kt:120]: Same comments as above, spacing and variable name

**Review by AlmasB — CHANGES_REQUESTED** [review]
Looks good, please have a look at the comments

**Comment by GeorgeJoLo** [comment]
I changed everything you mentioned.
I made a commit in my fork, but I didn't submit a new PR, but I believe this is enough.

Waiting for news!

**Comment by AlmasB** [comment]
Please check the CI output, copied for convenience:

```
Error:  Failed to execute goal org.jetbrains.kotlin:kotlin-maven-plugin:1.4.10:compile (compile) on project fxgl-animation: Compilation failure: Compilation failure: 
Error:  /home/runner/work/FXGL/FXGL/fxgl-animation/src/main/kotlin/com/almasb/fxgl/animation/AnimatedValue.kt:[111,17] Unresolved reference: Animated2D
Error:  /home/runner/work/FXGL/FXGL/fxgl-animation/src/main/kotlin/com/almasb/fxgl/animation/AnimatedValue.kt:[123,17] Unresolved reference: Animated2D
```

**Comment by GeorgeJoLo** [comment]
I corrected them!

**Review by AlmasB — APPROVED** [review]
Looks good, thanks!

**Comment by GeorgeJoLo** [comment]
Your welcome!

**Comment by AlmasB** [comment]
Closes #925

**Pull request merged**